### ISSUE TITLE
Allow alternatives to reflect-metadata for shimming reflect metadata API

### ISSUE THREAD

**Issue by iacobfred** [typegoose/typegoose#761]
## Describe what you need | want

Typegoose requires a shim for the reflect metadata API but does not need to require `reflect-metadata` (which is old and heavy) specifically. There are alternatives such as https://github.com/abraham/reflection

## Do you have already an idea for the implementation?

*Yes*. Interchangability of reflect metatdata shims is already implemented by TypeGraphQL; see https://github.com/MichalLytek/type-graphql/pull/1331#issuecomment-1242924189
```

**Comment by hasezoey** [comment]
i will consider it when we switch to ESM, because from what i can tell the mentioned library is ESM-only, which is a problem because typegoose will not switch to that any time soon, see #214 (from what i know CJS can be used in ESM, but ESM not in CJS without having a async function somewhere and typegoose currently is sync)

**Comment by hasezoey** [comment]
just had a user in discord having a problem where `reflect-metadata` would not set the custom options, but `` worked

so i will consider adding a way to change the reflection package used sooner

**Comment by hasezoey** [comment]
@KeithGillette did you confirm that only that reflect library is running? because `reflect-metadata` is a direct dependency of typegoose currently and theoretically always imported in typegoose

**Comment by KeithGillette** [comment]
No, @hasezoey, I did not realize that `reflect-metadata` was loaded directly by Typegoose. Other dependencies I've used calling the Reflect API have required that a Reflect shim be loaded in the project manually.
